Sources/CollectionsExtension.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  CollectionsExtension.swift
+//  EZSwiftExtensions
+//
+//  Created by Sanyal, Arunav on 3/5/17.
+//  Copyright © 2017 Goktug Yilmaz. All rights reserved.
+//
+
+import Foundation
+
+extension Collection {
+
+#if os(iOS)
+    // EZSE : A parralelized map for collections, operation is non blocking
+    public func pmap<R>(_ each: @escaping (Self.Iterator.Element) -> R) -> [R?] {
```

Comment:
I didn't like the name `pmap` for a method since It's not camel cased and it's not descriptive enough. Maybe `parallelizedMap()`?